client/tests/integration/permissions.rs

```diff
@@ -315,10 +315,9 @@ fn stored_vs_granted_token_payload() -> Result<()> {
     // Allow alice to mint mouse asset and mint initial value
     let mouse_asset = AssetId::new(asset_definition_id, mouse_id.clone());
     let allow_alice_to_set_key_value_in_mouse_asset = Grant::permission(
-        Permission::from_str_unchecked(
+        Permission::new(
             "CanSetKeyValueInUserAsset".parse().unwrap(),
-            // NOTE: Introduced additional whitespaces in the serialized form
-            &*format!(r###"{{ "asset_id" : "xor#wonderland#{mouse_id}" }}"###),
+            json!({ "asset_id": format!("xor#wonderland#{mouse_id}") }),
```

Comment:
yes, you can define `serde_json::Value` directly from a string

Reply:
I noticed that whitespaces were introduced in order to check that that payloads are compared by `serde_json::Value` equality rather than by JSON str comparison.

But... in this particular test there is no reason to do that. This test registers a single permission and doesn't check any comparisons.

There is another test below which actually grants the same permission with technically the same content, although they even differ in JSON:

```rust
let allow_alice_to_transfer_rose_1 = Grant::permission(
    Permission::new(
        "CanTransferUserAsset".parse().unwrap(),
        json!({ "asset_id": format!("rose#wonderland#{alice_id}") }),
    ),
    alice_id.clone(),
);

let allow_alice_to_transfer_rose_2 = Grant::permission(
    Permission::new(
        "CanTransferUserAsset".parse().unwrap(),
        // different content, but same meaning
        json!({ "asset_id": format!("rose##{alice_id}") }),
    ),
    alice_id,
);
```

1. They have different JSON payloads! Therefore, equality by `serde_json::Value` would fail to detect duplication.
2. Semantically, they have the same payloads, because it is still `AssetId`, but written in a different form.
3. And indeed, the second instruction fails as duplicate! Therefore, the comparison is made in executor-native types, not JSON.

I am _not sure_ why it works, but this test passes.

Reply:
previously tokens were compared as strings, not as JSON objects. `stored_vs_granted` test was introduce specifically to ensure that we compare it as JSON objects, so it's enough that you keep it only in there. Check #3853

Reply:
Got it, returned JSON with extra whitespaces